### ISSUE TITLE
Allow processing on an application whose window title changes frequently

### ISSUE THREAD

**Issue by Gitoffthelawn** [hluk/CopyQ#1553]
Currently, CopyQ can restrict rule processing to a window whose title matches a regex.  This works well.

But what about when the title of a window frequently changes?  Is there any way to match such a window?

Specifically, I'm thinking of Windows Explorer (aka Windows File Explorer), whose window title changes to the current folder (directory) every time a new folder is "explored" (at least in Win7).  Does CopyQ have any way to restrict processing to such windows, and if not, could such a feature be added?

**Comment by hluk** [comment]
Do you mean that there is no fixed text in the window title? Usually window titles end with application name.

**Comment by Gitoffthelawn** [comment]
> Do you mean that there is no fixed text in the window title? Usually window titles end with application name.

Correct.  For example, Microsoft, in their infinite wisdom, has no application name in the title bar of their most commonly used application: Windows Explorer.  :man_facepalming:

**Comment by hluk** [comment]
If you need this to identify app for automatic clipboard content handling in CopyQ, it might be possible to check if some special format is in the clipboard instead (use Ctrl+Shift+C to see the formats after copying).

**Comment by Gitoffthelawn** [comment]
Thanks.  I'll take a look at that possibility.

I'm also going to take a look at reliable methods to identify an application's window when the application's name is not in the title.  If you have any thoughts, I'm all ears.

**Comment by hluk** [comment]
There are `GetClassName` Windows functions which could give some reasonable results. But for CopyQ it returns `Qt5152QWindowIcon` - probably similar for all Qt 5 apps. This was just a quick automated test - I cannot test it on Windows machine right now.

Can you check if [this development build](https://ci.appveyor.com/project/hluk/copyq/builds/37206971/job/hkvilaec421m2bvk/artifacts) gives reasonable results? You can use the portable zip file so as not to mess with your current installation of CopyQ. To test it, you can use following command (bind a global shortcut, save, select a window and press the shortcut):

```
[Command]
Command="
    copyq:
    var cls = currentWindowClass();
    popup('Window Class (copied to clipboard)', cls)
    copy(cls)"
GlobalShortcut=meta+ctrl+w
Icon=\xf06e
IsGlobalShortcut=true
Name=Copy Window Class
```

[Here is how to add the command to CopyQ](http://copyq.readthedocs.io/en/latest/faq.html#how-to-load-shared-commands-and-share-them).

**Comment by Gitoffthelawn** [comment]
Yes, I will test.  It won't be immediately, but I will put it as close to the top of my list as I can for you!

**Comment by snowman** [comment]
```
Note [2023-04-08 18:19:02.503] <Server-2276>: Exception in command "Copy Window Class": ScriptError: ReferenceError: currentWindowClass is not defined
Note [2023-04-08 18:19:02.503] <Server-2276>:    
Note [2023-04-08 18:19:02.503] <Server-2276>:    --- backtrace ---
Note [2023-04-08 18:19:02.503] <Server-2276>:    %entry@eval code:2
Note [2023-04-08 18:19:02.503] <Server-2276>:    eval:var cls = currentWindowClass(...
Note [2023-04-08 18:19:02.503] <Server-2276>:    eval
Note [2023-04-08 18:19:02.503] <Server-2276>:    --- end backtrace ---
Note [2023-04-08 18:19:02.511] <Server-2276>: Command “Copy Window Class”
Note [2023-04-08 18:19:02.511] <Server-2276>:    Exit code: 4
Note [2023-04-08 18:19:02.511] <Server-2276>:    1. copyq: 
Note [2023-04-08 18:19:02.511] <Server-2276>:    2. var cls = currentWindowClass();
Note [2023-04-08 18:19:02.511] <Server-2276>:    3. popup('Window Class (copied to clipboard)', cls)
Note [2023-04-08 18:19:02.511] <Server-2276>:    4. copy(cls)
```
